src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/XmlDescriptor.php

```diff
@@ -331,18 +331,16 @@ private function getContainerDefinitionDocument(Definition $definition, $id = nu
 
         $serviceXML->setAttribute('class', $definition->getClass());
 
-        if (method_exists($definition, 'getFactoryMethod')) {
-            if ($definition->getFactoryClass(false)) {
-                $serviceXML->setAttribute('factory-class', $definition->getFactoryClass(false));
-            }
+        if ($definition->getFactoryClass(false)) {
+            $serviceXML->setAttribute('factory-class', $definition->getFactoryClass(false));
+        }
 
-            if ($definition->getFactoryService(false)) {
-                $serviceXML->setAttribute('factory-service', $definition->getFactoryService(false));
-            }
+        if ($definition->getFactoryService(false)) {
+            $serviceXML->setAttribute('factory-service', $definition->getFactoryService(false));
+        }
 
-            if ($definition->getFactoryMethod(false)) {
-                $serviceXML->setAttribute('factory-method', $definition->getFactoryMethod(false));
-            }
+        if ($definition->getFactoryMethod(false)) {
```

Comment:
@stof We're in the 2.8 branch and it requires **only** Symfony 2.8 (it doesn't work with 3.0). It must indeed be dropped when merging in 3.0.